src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -209,10 +230,17 @@ def get_or_create_project_uptime_subscription(
     )
 
     # Update status. This may have the side effect of removing or creating a
-    # remote subscription.
+    # remote subscription. When a new monitor is created we will ensure seat
+    # assignment, which may cause the monitor to be disabled if there are no
+    # available seat assignments.
     match status:
         case ObjectStatus.ACTIVE:
-            enable_project_uptime_subscription(uptime_monitor)
+            try:
+                enable_project_uptime_subscription(uptime_monitor, ensure_assignment=created)
+            except UptimeMonitorNoSeatAvailable:
+                # No need to do anything if we failed to handle seat
+                # assignment. The monitor will be created, but not enabled
```

Comment:
Should we mark it as disabled? Otherwise, the subscription checker will pick this up and create it

Reply:
`enable_project_uptime_subscription` will actually call `disable_project_uptime_subscription` if it cannot assign the seat.

I realize this logic is confusing and am open to maybe something more explicit?